From 6abc3b8f0b343eccac9af1c6009e8d3d1a2f973e Mon Sep 17 00:00:00 2001
From: lambdadroid <lambdadroid@gmail.com>
Date: Sat, 6 Jan 2018 14:48:44 +0100
Subject: [PATCH 14/24] vold: Accept Linux GPT partitions on external SD cards

ext4/f2fs partitions are usually created using the Linux partition
type on GUID partition tables, which is not recognized by vold.

Add the Linux partition type GUID to vold so ext4/f2fs partitions
on external SD cards with GPT are detected correctly.

Change-Id: I5e3947bb6b086eec4e5aa8a4825aaa93b7e9d99d
---
 model/Disk.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/model/Disk.cpp b/model/Disk.cpp
index a017387..9d323b4 100644
--- a/model/Disk.cpp
+++ b/model/Disk.cpp
@@ -78,6 +78,7 @@ static const unsigned int kMajorBlockExperimentalMin = 240;
 static const unsigned int kMajorBlockExperimentalMax = 254;
 
 static const char* kGptBasicData = "EBD0A0A2-B9E5-4433-87C0-68B6B72699C7";
+static const char* kGptLinuxFilesystem = "0FC63DAF-8483-4772-8E79-3D69D8477DE4";
 static const char* kGptAndroidMeta = "19A710A2-B3CA-11E4-B026-10604B889DCF";
 static const char* kGptAndroidExpand = "193D1EA4-B3CA-11E4-B075-10604B889DCF";
 
@@ -396,7 +397,8 @@ status_t Disk::readPartitions() {
                 if (++it == split.end()) continue;
                 auto partGuid = *it;
 
-                if (android::base::EqualsIgnoreCase(typeGuid, kGptBasicData)) {
+                if (android::base::EqualsIgnoreCase(typeGuid, kGptBasicData)
+                        || android::base::EqualsIgnoreCase(typeGuid, kGptLinuxFilesystem)) {
                     createPublicVolume(partDevice);
                 } else if (android::base::EqualsIgnoreCase(typeGuid, kGptAndroidExpand)) {
                     createPrivateVolume(partDevice, partGuid);
-- 
2.17.1

